src/lib.rs

```diff
@@ -151,6 +151,22 @@ struct HelperInner {
     consumer_done: bool,
 }
 
+/// Error type for `from_env` function.
+#[derive(Debug)]
+pub enum ErrFromEnv {
+    /// There isn't env var, that describes jobserver to inherit.
+    IsNotConfigured,
+    /// Cannot connect following this process's environment.
+    PlatformSpecific {
+        /// Error.
+        err: imp::ErrFromEnv,
+        /// Name of gotten env var.
+        env: &'static str,
+        /// Value of gotten env var.
+        var: String,
+    },
+}
+
```

Comment:
```suggestion
impl std::fmt::Display for ErrFromEnv {
    // ...
}

impl std::error::Error for ErrFromEnv {
    fn source(&self) -> Option<&dyn std::error::Error> {
        match self {
            ErrorFromEnv::PlatformSpecific { err } => Some(err),
            _ => None,
        }
    } 
}
```

Or you can use [`thiserror`](https://docs.rs/thiserror) for convenience, but since jobserver is used in cargo they might not want this.